### ISSUE TITLE
Time to first execution (TTFX)

### ISSUE THREAD

**Issue by maxchendt** [oxfordcontrol/Clarabel.jl#120]
on Julia 1.9 rc1,  there still needs 13 seconds for first run, and 0.3 seconds for second run, for one of my dataset.

I find that the package [`EfficientFrontier`](https://github.com/PharosAbad/EfficientFrontier.jl) cut down from 7.89 seconds ( Julia 1.8.5) to 0.13 seconds (1.9 rc1).  Lines like [`Base.precompile(Tuple{...})`](https://github.com/PharosAbad/EfficientFrontier.jl/blob/main/src/EfficientFrontier.jl#L36L43) may be helpful

**Comment by guilhermebodin** [comment]
Just as a reference, there is a big discussion about this in this issue https://github.com/jump-dev/JuMP.jl/issues/3192. They have made some precompilation statements for some solvers as a test. Maybe you could use some of the information even if not using the MOI interface.

**Comment by goulart-paul** [comment]
I made an initial attempt at this in [pg/precompile](https://github.com/oxfordcontrol/Clarabel.jl/tree/pg/precompile), just using SnoopPrecompile on the solver's direct interface (i.e. not via MOI).   

It doesn't seem to have made any improvement at all, even on julia v1.9, but I think I must be doing some really basic thing wrong.   I will revisit once I have had a closer look at the issue suggested by @guilhermebodin.

**Comment by maxchendt** [comment]
How about

```Julia
__precompile__()

.
.
.
Base.precompile(Tuple{Core.Typeof(Clarabel.Solver), Nothing})
Base.precompile(Tuple{Core.Typeof(Clarabel.Settings), Nothing})
Base.precompile(Tuple{Core.Typeof(Clarabel.solve!), Nothing})
setup!

```

**Comment by maxchendt** [comment]
How about

```Julia
__precompile__()

.
.
.
Base.precompile(Tuple{Core.Typeof(Clarabel.Solver), Nothing})
Base.precompile(Tuple{Core.Typeof(Clarabel.Settings), Nothing})
Base.precompile(Tuple{Core.Typeof(Clarabel.setup!), Solver{Float64}, Matrix{Float64}, Vector{Float64}, Matrix{Float64}, Vector{Float64}, Vector{SupportedCone}, Settings})
Base.precompile(Tuple{Core.Typeof(Clarabel.solve!), Solver{Float64}})

```

**Comment by goulart-paul** [comment]
> How about
```
__precompile__()
... etc
```

That also doesn't seem to improve things, and `@time using Clarabel` at launch (following a previously precompilation and restart) still takes about 5 seconds.   It also seems to have *increased* the precompilation time quite a lot with no real payoff.

**Comment by maxchendt** [comment]
May be you should add things like

```julia
Base.precompile(Tuple{typeof(computeCL!), Vector{sCL{Float64}}, Vector{Status}, Problem{Float64}, Settings{Float64}})
```
where the `computeCL!` is the core of the computation, but not directly expose to users.


I do not know what are the working horses of your functions.

**Comment by goulart-paul** [comment]
#122 has TTFX for the native solver interface (i.e. not MOI / JuMP) down to < 25ms.    Package load times are also now a lot shorter.

NB: testing described on that PR is all on julia 1.9.0-rc1 and the master branches of MOI and JuMP.

**Comment by goulart-paul** [comment]
This should be fixed in v0.5.0, released today.    I will close the issue but please reopen if there is still a problem.

For greatest improvement in load times you need julia v1.9 or higher.  It will still work with earlier versions though.
